client/hwctl/src/main.rs

```diff
@@ -27,7 +27,16 @@ use hwlib::{
 
 #[tokio::main]
 async fn main() -> Result<()> {
```

Comment:
This may be beyond the scope of this PR, but there are currently two ways for errors to be reported, either explicitly with the new `eprintln!` calls, or implicitly by returning `Err` from `main`. In the latter case, the error message will look different (and I think often uglier?), so it's probably best to write another function which catches all errors and outputs them in a unified way:
```rust
async fn main() -> ExitCode {
    let err = match run().err() {
        Some(err) => err,
        None => return ExitCode::SUCCESS,
    };
    eprintln!("{e}");
    ExitCode::FAILURE
}

fn run() -> Result<()> {
    // Everything currently in main
}

Reply:
I've decided to keep it in one function since otherwise, I feel like the `run` function will do almost nothing
isn't the purpose of `main` to call the functions in the correct order, handling errors if they appear?